src/Kdyby/Validator/ConstraintValidatorFactory.php

```diff
@@ -47,23 +49,16 @@ public function __construct(Nette\DI\Container $serviceLocator)
 	 */
 	public function getInstance(Constraint $constraint)
 	{
-		$className = $constraint->validatedBy();
+		$name = $constraint->validatedBy();
 
-		// Workaround for https://github.com/symfony/symfony/pull/16166.
-		if ($className === 'validator.expression') {
-			$className = 'Symfony\Component\Validator\Constraints\ExpressionValidator';
+		if (!isset($this->validators[$name])) {
+			// @todo: Deprecate or remove this?
+			$this->validators[$name] = $this->serviceLocator->createInstance($name);
+		} elseif (is_string($this->validators[$name])) {
+			$this->validators[$name] = $this->serviceLocator->get($this->validators[$name]);
```

Comment:
`$this->validator` should now contain names of the services. Look at the very end of the current diff.